prov/sockets/src/sock_av.c

```diff
@@ -386,15 +386,18 @@ static int sock_av_insertsym(struct fid_av *av, const char *node, size_t nodecnt
 		fmt = 0;
 	else 
 		fmt = offset;
-
+	
+	assert((hostlen-offset) < FI_NAME_MAX);
 	strncpy(base_host, node, hostlen - (offset));
 	var_port = atoi(service);
 	var_host = atoi(node + hostlen - offset);
 	
 	for (i = 0; i < nodecnt; i++) {
 		for (j = 0; j < svccnt; j++) {
-			sprintf(tmp_host, "%s%0*d", base_host, fmt, var_host + i);
-			sprintf(tmp_port, "%d", var_port + j);
+			ret = snprintf(tmp_host, FI_NAME_MAX, "%s%0*d", base_host, fmt, var_host + i);
+			assert(ret < FI_NAME_MAX);
+			ret = snprintf(tmp_port, FI_NAME_MAX,  "%d", var_port + j);
+			assert(ret < FI_NAME_MAX);
```

Comment:
My intention behind the assert is to fail when <code>(base_host, fmt, var_host + i)</code> is greater than FI_NAME_MAX (according to the definition writes FI_NAME_MAX-1 characters but returns the total number of the characters). But I can check the value and return error instead of an assert. Will update the PR then.